src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs

```diff
@@ -254,7 +255,7 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         &self,
         borrow_location: Location,
     ) -> bool {
-        let mut visited_locations = Vec::new();
+        let mut visited_locations = FxHashSet::default();
```

Comment:
Hm, I'd rather make this change in a separate PR which would be benchmarked to make sure this is actually an optimization (it looks like it would be but I'm not sure).